lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -532,15 +550,125 @@ defmodule ExUnit.Case do
+  @doc """
+  Registers a new describe attribute to be used during `ExUnit.Case` tests.
+
+  The attribute values will be available through `context.registered`.
+  Registered values are cleared after each `ExUnit.Case.describe/2` similar
+  to `@describetag`.
+
+  `Module.register_attribute/3` is used to register the attribute,
+  this function takes the same options.
+
+  ## Examples
+
+      defmodule MyTest do
+        use ExUnit.Case
+
+        ExUnit.Case.register_describe_attribute(__MODULE__, :describe_fixtures, accumulate: true)
+
+        describe "using custom attribute" do
+          @describe_fixtures :user
+          @describe_fixtures {:post, insert: false}
+
+          test "has attribute", context do
+            assert context.registered.describe_fixtures == [{:post, insert: false}, :user]
+          end
+        end
+
+        describe "custom attributes are cleared per describe" do
+          test "doesn't have attributes", context do
+            assert context.registered.describe_fixtures == []
+          end
+        end
+      end
+
+  """
+  @spec register_describe_attribute(env, name :: atom, opts :: keyword) :: :ok
+  def register_describe_attribute(env, name, opts \\ [])
+
+  def register_describe_attribute(%{module: mod}, name, opts) do
+    register_describe_attribute(mod, name, opts)
+  end
+
+  def register_describe_attribute(mod, name, opts)
+      when is_atom(mod) and is_atom(name) and is_list(opts) do
+    register_attribute(:ex_unit_registered_describe_attributes, mod, name, opts)
+  end
+
+  @doc """
+  Registers a new module attribute to be used during `ExUnit.Case` tests.
+
+  The attribute values will be available through `context.registered`.
+
+  `Module.register_attribute/3` is used to register the attribute,
+  this function takes the same options.
+
+  ## Examples
+
+      defmodule MyTest do
+        use ExUnit.Case
+
+        ExUnit.Case.register_module_attribute(__MODULE__, :module_fixtures, accumulate: true)
+
+        @module_fixtures :user
+        @module_fixtures {:post, insert: false}
+
+        test "using custom attribute", context do
+          assert context.registered.fixtures == [{:post, insert: false}, :user]
+        end
+
+        test "still using custom attribute", context do
+          assert context.registered.fixtures == [{:post, insert: false}, :user]
+        end
+      end
+
+  """
+  @spec register_module_attribute(env, name :: atom, opts :: keyword) :: :ok
+  def register_module_attribute(env, name, opts \\ [])
+
+  def register_module_attribute(%{module: mod}, name, opts) do
+    register_module_attribute(mod, name, opts)
+  end
+
+  def register_module_attribute(mod, name, opts)
+      when is_atom(mod) and is_atom(name) and is_list(opts) do
+    register_attribute(:ex_unit_registered_module_attributes, mod, name, opts)
+  end
+
+  defp register_attribute(type, %{module: module}, name, opts) do
```

Comment:
I think this clause is never executed :)

Reply:
Good catch